lib/portlayer/exec/exec.go

```diff
@@ -152,8 +158,36 @@ func eventCallback(ie events.Event) {
 
 			}
 		}
+		return
 	}
+	// check container registered event if this container is not found in container cache
+	switch ie.String() {
+	case events.ContainerRegistered:
+		moref := new(types.ManagedObjectReference)
+		if ok := moref.FromString(ie.Reference()); !ok {
+			log.Errorf("Failed to get event VM mobref: %s", ie.Reference())
+			return
+		}
+		var vm mo.VirtualMachine
+
+		// current attributes we care about
+		attrib := []string{"config", "runtime.powerState", "summary", "resourcePool"}
 
+		// populate the vm properties
+		ctx := context.Background()
+		if err := eventSession.RetrieveOne(ctx, *moref, attrib, &vm); err != nil {
+			log.Errorf("Failed to query registered vm object %s: %s", ie.Reference(), err)
+			return
+		}
+		if *vm.ResourcePool != Config.ResourcePool.Reference() {
```

Comment:
This seems to be very specific to VM objects. Maybe we should move this "fix up" to the VirtualMachine class (via a WaitForResult or similar method), and not do it here? Adding these two parameters really makes this function very confusing, and ends up with WaitForResult doing to much imho.